staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go

```diff
@@ -695,7 +695,7 @@ func (w *watchCache) isIndexValidLocked(index int) bool {
 func (w *watchCache) getAllEventsSinceLocked(resourceVersion uint64, key string, opts storage.ListOptions) (*watchCacheInterval, error) {
 	_, matchesSingle := opts.Predicate.MatchesSingle()
 	matchesSingle = matchesSingle && !opts.Recursive
-	if opts.SendInitialEvents != nil && *opts.SendInitialEvents {
+	if isListWatchRequest(opts) {
```

Comment:
This is a behavior change - we weren't checking for AllowWatchBookmarks before.

@p0lyn0mial - do you remember if that was intentional?